R/build-reference.R

```diff
@@ -169,12 +169,11 @@ build_reference <- function(pkg = ".",
   check_character(topics, allow_null = TRUE)
 
   if (document != "DEPRECATED") {
-    lifecycle::deprecate_warn(
-      "1.4.0",
+    lifecycle::deprecate_stop(
+      "1.6.0",
       "build_site(document)",
-      details = "build_site(devel)"
+      "build_site(devel)"
```

Comment:
This gives a better lifecycle warning, so this change is worth keeping!

Before

```
Error:
! The `document` argument of `build_site()` was deprecated in <NA> 2.1.0 and is now defunct.
ℹ build_site(devel)
```

This PR
```r
Error:
! The `document` argument of `build_site()` was deprecated in <NA> 2.1.0 and is now defunct.
ℹ Please use the `devel` argument instead.
```